src/routes/Profile/Profile.tsx

```diff
@@ -89,14 +127,59 @@ const Profile: React.FunctionComponent = () => {
           claimedAt: b.variant === variant ? Date.now().valueOf() : b.claimedAt,
         }));
       });
-    }, 1500);
+    } catch (err) {
+      setClaimButtonModes((prev) => ({
+        ...prev,
+        [variant]: 'claimError',
+      }));
+    }
   }
 
-  function handleOnClaimBulkClick(variants: BadgeVariant[]) {
+  async function handleOnClaimBulkClick(variants: BadgeVariant[]) {
```

Comment:
this needs to be properly qa'd and also potentially worth doing some more elaborate e2e testing using this functionality as a case study to make the ui dev approach more TDD

Reply:
setting e2e is not really that straight forward for the crypto world. We need to set a proper infra. But I am in the process of testing with @ioanaroceanu2000